pkg/policy/policy_controller.go

```diff
@@ -568,8 +571,61 @@ func (pc *PolicyController) updateUR(policy kyverno.PolicyInterface, urList []*u
 		return
 	}
 
+	pc.generateTargets(policy)
 	updateUR(pc.kyvernoClient, policy.GetName(), urList, pc.log.WithName("updateUR"))
+}
+
+// generateTargets ...
+func (pc *PolicyController) generateTargets(policy kyverno.PolicyInterface) {
+	for _, rule := range policy.GetSpec().Rules {
+		if !rule.IsGenerateExisting() {
+			continue
+		}
+
+		// generateTriggers gets the generate triggers for given matchresource
+		// rule
+		triggers := generateTriggers(pc.nsLister, rule, pc.log)
+		for _, trigger := range triggers {
+			trigger := trigger //pin it
+			ur := newUR(policy, trigger, urkyverno.Generate)
+
+			// here trigger is the namespace for namespace matched kind policy
+			triggerResource, err := common.GetResource(pc.client, ur.Spec, pc.log)
+			if err != nil {
+				pc.log.WithName(rule.Name).Error(err, "failed to get trigger resource")
+			}
 
+			// check if the target resource exists in given namespace, as we don't
+			// want to create new updaterequest for same namespace
+			_, err = pc.client.GetResource(rule.Generation.APIVersion, rule.Generation.Kind, trigger.Name, rule.Generation.Name)
```

Comment:
Currently UR is created per policy, can you just check the UR labels? Why get the target resource here?

Reply:
As i am getting all the  matched namespace as a trigger including the one which already has the UR created for ( i.e. target resource exists) , in such cases i wanted to verify before creating the UR to trigger the rule.

Reply:
Updated to use UR Labels to verify the UR exists